MekHQ/src/mekhq/gui/preferences/JWindowPreference.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2019 The MegaMek Team. All rights reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.gui.preferences;
+
+import mekhq.preferences.PreferenceElement;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ComponentEvent;
+import java.awt.event.ComponentListener;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowStateListener;
+import java.lang.ref.WeakReference;
+
+public class JWindowPreference extends PreferenceElement implements WindowStateListener, ComponentListener {
+    private final WeakReference<Window> weakRef;
+    private int width;
+    private int height;
+    private int screenX;
+    private int screenY;
+    private boolean isMaximized;
+
+    public JWindowPreference(Window window) {
+        super (window.getName());
+
+        try {
+            this.width = window.getWidth();
+            this.height = window.getHeight();
+            this.screenX = window.getLocationOnScreen().x;
+            this.screenY = window.getLocationOnScreen().y;
+
+            if (window instanceof JFrame) {
+                this.isMaximized = (((JFrame)window).getState() & Frame.MAXIMIZED_BOTH) == Frame.MAXIMIZED_BOTH;
+            } else {
+                this.isMaximized = false;
+            }
+        } catch (Exception ignored) {
```

Comment:
which exception are we ignoring?

Reply:
On JDialogs, getLocationOnScreen() can throw as we are usually asking for this data before the dialog is visible. Dialogs can't be maximized either so if it throws when asking for getLocationOnScreen, then all is fine. The user will either don't move the dialog, so next launch will be again in the default location, or the user may move the dialog in that case we will get the updated location on screen from the componentMoved event. I'll add a comment about this.